eth/handler.go

```diff
@@ -188,7 +188,10 @@ func NewProtocolManager(config *params.ChainConfig, checkpoint *params.TrustedCh
 		}
 		return n, err
 	}
-	manager.blockFetcher = fetcher.NewBlockFetcher(blockchain.GetBlockByHash, validator, manager.BroadcastBlock, heighter, inserter, manager.removePeer)
+	hasBlock := func(h common.Hash) bool {
+		return blockchain.GetBlockByHash(h) != nil
+	}
```

Comment:
I don't really follow why you are using GetBlockByHash instead of HasBlock? The latter should be a lot lighter as it doesn't need to pull the data from disk. Am I missing something?

Reply:
You mean `HasBlock(hash common.Hash, number uint64)` ? 

I don't have the number. But you're right, all cases where we invoke this, we do have the number accessible. Should I change it? 
Secondly, it has slightly different semantics, and e.g. doesn't add it to the internal caches. I wasn't sure about the effects of changing that.

Reply:
The lighter method is txpool.Has, is it?

Reply:
Yes, please change it. Using the number avoids an additional index read to find the number.

Reply:
Ok, done